aiohttp/web_urldispatcher.py

```diff
@@ -414,6 +414,65 @@ def __repr__(self) -> str:
         return f"<PlainResource {name} {self._path}>"
 
 
+class GroupPlainResource(Resource):
+    def __init__(self) -> None:
+        super().__init__()
+        self._resources: Dict[str, Resource] = {}
+
+    @property
+    def canonical(self) -> str:
+        return "/"
+
+    def freeze(self) -> None:
+        self._routes.clear()
+        for resource in self._resources.values():
+            self._routes.extend(cast(Iterator[ResourceRoute], resource))
```

Comment:
Why is the `cast()` needed? The definition just above says this should be a `Resource`, so how is it `Iterator[ResourceRoute]`?

Reply:
https://github.com/aio-libs/aiohttp/runs/4636862493?check_suite_focus=true

```
Run make mypy
mypy
aiohttp/web_urldispatcher.py:429: error: Argument 1 to "iter" has incompatible type "Resource"; expected "Iterable[ResourceRoute]"  [arg-type]
aiohttp/web_urldispatcher.py:429: note: Following member(s) of "Resource" have conflicts:
aiohttp/web_urldispatcher.py:429: note:     Expected:
aiohttp/web_urldispatcher.py:429: note:         def __iter__(self) -> Iterator[ResourceRoute]
aiohttp/web_urldispatcher.py:429: note:     Got:
aiohttp/web_urldispatcher.py:429: note:         def __iter__(self) -> Iterator[AbstractRoute]
Found 1 error in 1 file (checked 124 source files)
make: *** [Makefile:89: mypy] Error 1
Error: Process completed with exit code 2.
```

Reply:
Fixed: #6460